LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift

```diff
@@ -0,0 +1,317 @@
+            } else {
+                return Self.defaultWalshInsulinModelDuration
+            }
+        }
+        set {
+            precondition(Self.validWalshModelDurationRange.contains(newValue))
+            insulinModelSettings = .walsh(WalshInsulinModel(actionDuration: newValue))
+        }
+    }
+
+    public init(insulinModelSettings: InsulinModelSettings, insulinSensitivitySchedule: InsulinSensitivitySchedule?) {
+        self._insulinModelSettings = Published(wrappedValue: insulinModelSettings)
+        self.insulinSensitivitySchedule = insulinSensitivitySchedule ?? Self.defaultInsulinSensitivitySchedule
+    }
+}
+
+public struct InsulinModelSelection: View, HorizontalSizeClassOverride {
+
+    @ObservedObject var viewModel: InsulinModelSelectionViewModel
+    var glucoseUnit: HKUnit
+    var supportedModelSettings: SupportedModelSettings
+    var appName: String
+
+    let chartManager: ChartsManager = {
+        let chartManager = ChartsManager(
+            colors: .default,
+            settings: .default,
+            axisLabelFont: .systemFont(ofSize: 12),
+            charts: [InsulinModelChart()],
+            traitCollection: .current
+        )
+
+        chartManager.startDate = Calendar.current.nextDate(
+            after: Date(),
+            matching: DateComponents(minute: 0),
+            matchingPolicy: .strict,
+            direction: .backward
+        ) ?? Date()
+
+        return chartManager
+    }()
+
+    @Environment(\.dismiss) var dismiss
+    
+    public init(
+        viewModel: InsulinModelSelectionViewModel,
+        glucoseUnit: HKUnit,
+        supportedModelSettings: SupportedModelSettings,
+        appName: String
+    ) {
+        self.viewModel = viewModel
+        self.glucoseUnit = glucoseUnit
+        self.supportedModelSettings = supportedModelSettings
+        self.appName = appName
+    }
+
+    public var body: some View {
+        NavigationView {
+            List {
+                Section {
+                    SettingDescription(
+                        text: insulinModelSettingDescription,
+                        informationalContent: {
+                            // TODO: Implement informational content
+                            Text("Not implemented")
+                        }
+                    )
+                    .padding(4)
+                    .padding(.top, 4)
+
+                    VStack {
+                        InsulinModelChartView(
+                            chartManager: chartManager,
+                            glucoseUnit: glucoseUnit,
+                            selectedInsulinModelValues: selectedInsulinModelValues,
+                            unselectedInsulinModelValues: unselectedInsulinModelValues,
+                            glucoseDisplayRange: endingGlucoseQuantity...startingGlucoseQuantity
+                        )
+                        .frame(height: 170)
+
+                        InsulinModelSettingsRow(
```

Comment:
Looks like somewhere along the way there may have been a merge issue here, I believe @rickpasetto refactored this checkmark row component, and it was renamed

Reply:
I just re-copy-pasted in. If there are any other screens that I moved from Loop->LoopKit that have been modified within the last week, I may not have the latest copy of the changes. I think this is basically the only one, though, aside from the color changes @nhamming is making